lib/containers/common.pm

```diff
@@ -326,4 +328,23 @@ sub check_containers_connectivity {
     assert_script_run "$runtime kill $container_name";
 }
 
+sub switch_cgroup {
+    my ($self, $version) = @_;
+
+    my $setting = ($version == 2) ? 1 : 0;
+
+    record_info "cgroup v$version", "Switching to cgroup v$version";
+    if (is_transactional) {
+        add_grub_cmdline_settings("systemd.unified_cgroup_hierarchy=$setting", update_grub => 0);
+        assert_script_run('transactional-update grub.cfg');
+        process_reboot(trigger => 1);
+    } else {
+        add_grub_cmdline_settings("systemd.unified_cgroup_hierarchy=$setting", update_grub => 1);
+        power_action('reboot', textmode => 1);
+        $self->wait_boot(bootloader_time => 360);
+    }
+
+    assert_script_run("cat /proc/cmdline | grep systemd\.unified_cgroup_hierarchy=$setting");
```

Comment:
```suggestion
    validate_script_output("cat /proc/cmdline", sub { m/systemd\.unified_cgroup_hierarchy=$setting/ } );
```
`validate_script_output` is IMHO better than `assert_script_run` because it shows the output in the error case instead of just silently failing.

Reply:
Wouldn't it need the `select_serial_terminal;` from the code where I took it?

Reply:
It fails:
https://openqa.suse.de/tests/12868890#step/rootless_podman/64